frontend/src/DialectReportView.tsx

```diff
@@ -22,27 +28,82 @@ export const DialectReportView = ({
   }, [reportData]);
 
   const filteredData = useMemo(() => {
-    let filteredData = reportData;
-    if (params.getAll("language").length) {
-      const filteredArray = Array.from(
+    const filteredReportData = { ...reportData };
+    let filteredOtherImplementationsData: Record<string, Implementation> = {};
+    const selectedLanguages = params.getAll("language");
+
+    if (selectedLanguages.length > 0) {
+      const filteredReportArray = Array.from(
         reportData.implementations.entries(),
       ).filter(([, data]) =>
-        params.getAll("language").includes(data.metadata.language),
+        selectedLanguages.includes(data.metadata.language),
+      );
+      const filteredReportImplementationsMap = new Map(filteredReportArray);
+      filteredReportData.implementations = filteredReportImplementationsMap;
+      if (allImplementationsData) {
+        filteredOtherImplementationsData = filterOtherImplementations(
+          allImplementationsData,
+          selectedLanguages,
+          filteredReportImplementationsMap,
+        );
+      }
+    } else {
+      const filteredReportArray = Array.from(
+        reportData.implementations.entries(),
       );
-      filteredData = { ...reportData, implementations: new Map(filteredArray) };
+      const filteredReportImplementationsMap = new Map(filteredReportArray);
+      if (allImplementationsData) {
+        filteredOtherImplementationsData = filterOtherImplementations(
+          allImplementationsData,
+          languages,
+          filteredReportImplementationsMap,
+        );
+      }
     }
-    return filteredData;
-  }, [reportData, params]);
+    return { filteredReportData, filteredOtherImplementationsData };
```

Comment:
Maybe it would be easier to read if this memo was split into two? It's pretty complex now